node/node.go

```diff
@@ -1407,6 +1525,45 @@ func makeNodeInfo(
 	return nodeInfo, err
 }
 
+func makeSeedNodeInfo(
+	config *cfg.Config,
+	nodeKey p2p.NodeKey,
+	genDoc *types.GenesisDoc,
+	state sm.State,
+) (p2p.NodeInfo, error) {
+	nodeInfo := p2p.NodeInfo{
+		ProtocolVersion: p2p.NewProtocolVersion(
+			version.P2PProtocol, // global
+			state.Version.Consensus.Block,
+			state.Version.Consensus.App,
```

Comment:
I'm not too sure how full nodes handle this, but AFAICS, state is never updated. I can foresee then that there might be a problem if there is a change to the Block and/or App version where the seed node doesn't get updated and can thus no longer communicate with the block chain. I guess this would mean that the operator would also need to stop their seed node and update their genesis.